R/post-body.R

```diff
@@ -24,3 +30,27 @@ parseBody <- function(body){
   }
   ret
 }
+
+#' @importFrom utils URLdecode
+#' @noRd
+parseMultipart <- function(body){
+  # This is not as generic as perhaps it should be, but should be relatively
+  # easy to extend if more binary formats are required
+  # Is there data in the request?
+  if (is.null(body) || length(body) == 0 || body == "") {
+    return(list())
+  }
+  # This is something odd with webutils::parse_multipart. For some reason it
+  # does not work when the boundary is set to the content type as per the
+  # webutils docs.
+  boundary <- "-----------------"
+  parsed_binary <- webutils::parse_multipart(body, boundary)
+  file_name <- parsed_binary[[1]][[1]]
+  file_data <- parsed_binary[[1]][[2]]
```

Comment:
I'm  not sure yet. The list indexes used here are not the best way of accessing the parsed content. This looks better (`myfile` is a name assigned by `webutils::parse_multipart()`)

    parsed_binary$myfile$filename
    parsed_binary$myfile$value

For readability I've now implemented it as named lists, and have yet to figure out yet how to parse multiple files.